tests/Resources/ti.ui.tabgroup.test.js

```diff
@@ -379,6 +374,101 @@ describe('Titanium.UI.TabGroup', function () {
 			tabGroup.addTab(tabB);
 			tabGroup.open();
 		});
+
+		describe('#close()', () => {
+			it('is a Function', () => {
+				tabGroup = Ti.UI.createTabGroup();
+
+				should(tabGroup).have.a.property('close').which.is.a.Function();
+			});
+
+			it('returns a Promise', finish => {
+				const tabA = Ti.UI.createTab({
+					title: 'Tab A',
+					window: Ti.UI.createWindow()
+				});
+				const tabB = Ti.UI.createTab({
+					title: 'Tab B',
+					window: Ti.UI.createWindow()
+				});
+				tabGroup = Ti.UI.createTabGroup();
+				tabGroup.addTab(tabA);
+				tabGroup.addTab(tabB);
+
+				const openPromise = tabGroup.open();
+				openPromise.then(() => {
+					const result = tabGroup.close();
+					result.should.be.a.Promise();
+					result.then(() => finish()).catch(e => finish(e));
+				}).catch(e => finish(e));
+			});
+
+			it('called on unopened Window rejects Promise', finish => {
+				tabGroup = Ti.UI.createTabGroup();
+				const tabA = Ti.UI.createTab({
+					title: 'Tab A',
+					window: Ti.UI.createWindow()
+				});
+				tabGroup.addTab(tabA);
+
+				const result = tabGroup.close();
+				result.should.be.a.Promise();
+				result.then(() => finish(new Error('Expected #close() to be rejected on unopened TabGroup'))).catch(_e => finish());
+			});
+
+			it('called twice on Window rejects second Promise', finish => {
+				tabGroup = Ti.UI.createTabGroup();
+				const tabA = Ti.UI.createTab({
+					title: 'Tab A',
+					window: Ti.UI.createWindow()
+				});
+				tabGroup.addTab(tabA);
+
+				tabGroup.open().then(() => {
+					tabGroup.close().then(() => {
```

Comment:
<!--
  0 failure: 
  3 warning:  tests/Resources/t..., tests/Resources/t..., tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.ui.tabgroup.test.js;
  Line: 428;
-->

- :warning: tests/Resources/ti.ui.tabgroup.test.js line 428 – Avoid nesting promises. (promise/no-nesting)
- :warning: tests/Resources/ti.ui.tabgroup.test.js line 428 – Avoid nesting promises. (promise/no-nesting)
- :warning: tests/Resources/ti.ui.tabgroup.test.js line 428 – Each then() should return a value or throw (promise/always-return)